src/foam/core/stdlib.js

```diff
@@ -338,19 +338,31 @@ foam.LIB({
 })();
 
 
-foam.LIB({
-  name: 'foam.Number',
-  methods: [
-    function isInstance(o) { return typeof o === 'number'; },
-    function clone(o) { return o; },
-    function equals(a, b) { return a === b; },
-    function compare(a, b) {
-      return ( b === null || b === undefined ) ? 1 :
-        a < b ? -1 : a > b ? 1 : 0;
-    },
-    function hashCode(n) { return n & n; }
-  ]
-});
+(function() {
+  var bufForHash = new ArrayBuffer(8);
+  var floatArrayForHash = new Float64Array(bufForHash);
+  var intArrayForHash = new Int32Array(bufForHash);
+
+  foam.LIB({
+    name: 'foam.Number',
+    methods: [
+      function isInstance(o) { return typeof o === 'number'; },
+      function clone(o) { return o; },
+      function equals(a, b) { return a === b; },
+      function compare(a, b) {
+        return ( b === null || b === undefined ) ? 1 :
+            a < b ? -1 : a > b ? 1 : 0;
+      },
+      function hashCode(n) {
```

Comment:
Personally I would find it a bit cleaner to build the closure here just for the hashCode function.

But it's not a big deal.